torch/csrc/jit/interpreter.cpp

```diff
@@ -434,6 +444,13 @@ struct CodeImpl {
   void insertInstruction(OpCode op, int64_t X = 0, uint64_t N = 0) {
     instructions_.emplace_back(op, X, N);
     instructions_source_.emplace_back(current_node_);
+    if (torch::autograd::profiler::hasCallbacks()) {
```

Comment:
the problem is that insertInstruction gets called only once when the module is first executed, and not on each forward(). So it's possible that profiler was initialized only after.

I also feel that if profiling is on, storing shared_ptr<CodeImpl> (or some separate debug data struct) might be preferable than proactively stringifying all instructions.

Reply:
@dzhulgakov, regarding when the profiler is initialized comment: We can record debug information about graph nodes unconditionally. This still keeps events logging the information needed conditional.

Regarding your second point: sure. We can store shared_ptr to Node perhaps.